core/WFCORE-1427_ Convenient_declaration_of_server_suspend_timeout.adoc

```diff
@@ -0,0 +1,235 @@
+= Title
+:author:            Brian Stansberry
+:email:             brian.stansberry@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+In WildFly 10 we introduced the server suspend / graceful shutdown feature. This included a new "suspend" management operation, plus a "timeout" param to the existing "shutdown" operation. If timeout was set it would result in the server waiting up to the timeout to suspend before continuing with shutdown, thus providing a 'graceful shutdown'.
+
+However, there are other ways to trigger a server to stop handling requests besides the low level "shutdown" operation and the high level CLI "shutdown" command. This RFE is to add a "suspend-timeout" parameter to those, thus making it easier to get a graceful shutdown.
+
+Specifically, users can tell servers to "reload" in various ways, with a reload functioning effectively like a shutdown + restart. And then in a managed domain there are "restart-servers" ops at the domain and server group levels.
+
+For all of these it is possible for the user to get graceful behavior by first doing a "suspend" on the relevant servers before doing the reload / restart. But it's more user friendly to allow the user to provide a timeout similarly to what we do with shutdown.
+
+== Issue Metadata
+
+=== Issue
+
+* {issue-base-url}/EAP7-601[EAP7-601]
+* {issue-base-url}/WFCORE-3724[WFCORE-3724]
+* {issue-base-url}/WFCORE-1427[WFCORE-1427]
+
+=== Related Issues
+
+* {issue-base-url}/WFCORE-1632[WFCORE-1632]
+* {issue-base-url}/WFCORE-1917[WFCORE-1917]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+* mailto:yborgess@redhat.com[Yeray Borges]
+* mailto:jdenise@redhat.com[Jean-Francois Denise]
+
+=== QE Contacts
+
+* mailto:{jstourac@redhat.com}[{Jan Stourac}]
+
+=== Affected Projects or Components
+* WildFly Core kernel
+* WildFly Core CLI
+* HAL
+
+=== Other Interested Projects
+* N/A
+
+== Requirements
+
+=== Hard Requirements
+* All low level management operations that result in a server stopping handling of requests should accept a 'suspend-timeout' parameter that controls how long the server should wait for in-flight requests to complete before proceeding with the stop.
+** Meaning of the suspend-timeout should be the same as the 'timeout' param currently available for the 'shutdown' operation. A value of less than 0 means wait indefinitely, a value of 0 means don't wait, a value larger than zero means the server will wait up to this many seconds for all active requests to finish. Default value is 0. Value is in seconds.
+** Operations like 'shutdown' that already have a 'timeout' param are excepted from this requirement; no need to add a semantically equivalent 'suspend-timeout'.
```

Comment:
The idea of using `suspend-timeout` instead of `timeout` was because the attribute being introduced was not really a "shutdown servers timeout". It was a timeout for suspending the servers prior a subsequent action, which could be a restart. There was already a timeout in the standalone shutdown operation and the idea was to avoid confusion with it.

Reply:
My instinct without a ton of thought is that for a 'shutdown' op or command the 'timeout' param should be deprecated and a 'suspend-timeout' added.  For a 'suspend' op the name of the op provides context about what the timeout is. For 'shutdown', 'reload' etc the meaning is not clear at all and to met the intuitive meaning is that it applies somehow to the entire operation not just the suspend part.

Reply:
Generally, I agree and I think it is a good approach. But to be honest - I would prefer to have same name of the attribute also for `suspend` op/cmd. Reason is when one uses different commands, these two attribute names might be easily exchangable and potentionally confusing (one command uses `timeout`, others `suspend-timeout`). So maybe the best option is to use just `suspend-timeout` for all the operations, wdyt?

Reply:
I haven't taken into account this change yet in the new analysis doc version, but to me, it makes sense the deprecation of 'timeout' and the uses of 'suspend-timeout' instead for all of these operations including the suspend op. It makes a consistency between all of them.

Reply:
@jstourac I reviewed this part, now the analysis includes a note to deprecate the timeout parameter and to use suspend-timeout instead. Please, take a look at the current document version and let us know your concerns, thanks.